### ISSUE TITLE
Dashboard Grades

### ISSUE THREAD

**Issue by BGHDDevelopment** [ksucpea/bettercanvas#22]
Hey!

Dashboard grades seem to be glitched on my end again. 
They only seem to appear for classes that do not have posted grades
<img width="199" alt="chrome_rI2HqXEjYw" src="https://user-images.githubusercontent.com/16854333/224783579-765056e4-9224-4a56-85f9-feb8e7414945.png">
For classes that do have grades the overlay does not appear at all:
<img width="869" alt="chrome_1aPEpQ25rb" src="https://user-images.githubusercontent.com/16854333/224783679-ee9c6ecc-bfd7-4202-95f3-8e1116870e76.png">

**Comment by ksucpea** [comment]
Hey there, sorry I'm responding so late. Any chance you have more than 10 classes on your dashboard?

**Comment by BGHDDevelopment** [comment]
All good. There are 8 active pinned classes on the dashboard. In terms of in the class list and not pinned, my school does not remove them so over 30.

**Comment by ksucpea** [comment]
I think right now it only get 10 classes at a time so that could be why. I'll change it to get more classes in the next update and we'll see if that works

**Comment by BGHDDevelopment** [comment]
Sounds good. Is there any debug mode that I can activate to see in console what it's pulling? Would be helpful for any future debugging.

**Comment by ksucpea** [comment]
IF you go into the devtools under the network tab there should be a fetch request for "api/v1/courses?enrollment_state=active&include[]=total_scores&include[]=current_grading_period_scores", if you can find this you can preview the data its getting and see the classes it's getting grades for, can you check to see if it's getting the right classes?
[1](https://user-images.githubusercontent.com/59516750/226203467-28bd1e60-3dad-496e-be07-5a7a1eb0062d.png)

**Comment by BGHDDevelopment** [comment]
Nope pulling courses that are not pinned. I would think if there is a way to just grab stars/pinned classes that would work best to fix the overall issue. At least my school's system automatically adds/removes pins dependent on your active classes.

**Comment by ksucpea** [comment]
I agree, that would be better. I specified that the enrollment_state is active but maybe theres a different state I can try. If not I'll just get more classes

**Comment by ksucpea** [comment]
Not sure what you mean, like in terms of efficiency if I left that part out?

**Comment by ksucpea** [comment]
Hi again, didn't see your other comment, but I just updated it for Firefox, Chrome will be out in a day or two. Pretty sure I fixed the issue by just getting more classes, but let me know if you still have problems

**Comment by BGHDDevelopment** [comment]
Just tested it, seems to grab the majority now still missing a few more though. It may just be due to how my university refuses to remove old courses from the active list.

**Comment by ksucpea** [comment]
Dang, well I'll see if I can find a different way. The limit is now set to 30 courses but I think I could make it even more if I can't find anything
